### PR TITLE
Preparing for release: api and documentation

### PR DESCRIPTION
**Pull request by aspiwack**
I've tried to make the Haddock look good, basically.

### CODE REVIEW

File: src/Capability.hs [hunk 1]
```diff
@@ -51,6 +51,29 @@
...
+-- @
+-- f :: HasReader "config" Config m => …
+-- @
+--
+-- over and over again.
+--
+-- To avoid this, each capability comes with a /functional/—here
+-- @HasReader\'@—variant (in this terminology @HasReader@ is
+-- /relational/). Where the type is deduced from the capability's name. The
+-- mapping from name to type is done with the @'Capability.TypeOf.TypeOf'@
```
Comment (gdeest) [inline src/Capability.hs]: Is it `TypeOf` or `TypeOf'` ? (You link to `TypeOf'`, but the example below shows `TypeOf`).

File: src/Capability/Derive.hs [hunk 2]
```diff
@@ -9,21 +9,41 @@
...
--- current context via the @t@ wrapper, retaining the list @cs@ of capabilities.
-context ::
+-- | @'derive' \@t \@derived \@ambient act@ is used to run @act@ when the
+-- capabilities required by @act@ are not necessarily the same as those
+-- available.
+--
+-- @'derive' \@t \@derived \@ambient act@ runs @act@ by providing both the
+-- capabilities in @derived@ and @ambient@. The difference is that @ambient@
+-- capabilities are assumed to be available, whereas @derived@ instances are
+-- provided by @t@.
```
Comment (aherrmann) [inline src/Capability/Derive.hs]: There's a bit of duplication between those sentences. Maybe we could just drop the first sentence?

Comment (aspiwack) [reply src/Capability/Derive.hs]: Not sure. It's really on purpose. The first paragraph answers the question of: “why”, and the second paragraph answers “how”.

Is it grating to read? I'm really, really ok with duplication. Unless it really sounds horrible :slightly_smiling_face: .

Comment (aherrmann) [reply src/Capability/Derive.hs]: I see, okay with that spelled out I can see how those two sentences serve different purposes. I personally find it looks a bit scary when rendered. Together with the signature we're looking at three lines of complicated looking code without an introduction in plain English.

What do you think of this as a first sentence instead?
```
-- | Runs an action that requires additional capabilities.
```

File: src/Capability/Derive.hs [hunk 3]
```diff
@@ -9,21 +9,41 @@
...
+--
+-- 'derive' assumes that @t@ is a newtype defined in the form:
+--
+-- @
+-- newtype T m a = T (m a)
+-- @
+--
+-- Then 'derive' uses type-class instances for `T` to provide for each of the
+-- capability in @derived@.
```
Comment (aherrmann) [inline src/Capability/Derive.hs]:
```suggestion
-- capabilities in @derived@.
```

File: src/Capability/Derive.hs (hunk 2 above)
Comment (aherrmann) [reply src/Capability/Derive.hs]: I've changed the first sentence.

**Comment by aherrmann** [comment]
I've rebased on master and force-pushed. Hopefully that fixes CI.
